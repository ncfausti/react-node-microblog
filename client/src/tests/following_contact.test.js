/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import {
  fireEvent, render, waitFor,
} from '@testing-library/react';
import FollowingContact from '../components/home/following_contact';

test('test button click', async () => {
  const mock = jest.fn().mockImplementation(() => {});
  const { getByRole } = render(<FollowingContact username='feng' userid={31} unfollow={mock}/>);
  fireEvent.click(getByRole('button', { name: 'Unfollow' }));
  await waitFor(() => {
    expect(mock).toHaveBeenCalled();
  });
});

test('Page matches snapshot', () => {
  const component = renderer.create(<FollowingContact username='feng' userid={31}/>);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
