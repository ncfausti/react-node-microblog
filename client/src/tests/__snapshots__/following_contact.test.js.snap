// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page matches snapshot 1`] = `
<div
  className="following-contact"
>
  <span
    style={
      Object {
        "fontWeight": "bold",
      }
    }
  >
    feng
  </span>
  <button
    onClick={[Function]}
    style={
      Object {
        "border": "none",
        "borderRadius": "3px",
        "float": "right",
        "fontSize": "11px",
      }
    }
  >
    Unfollow
  </button>
</div>
`;
